internal/cli/atlas/alerts/settings/settings.go

```diff
@@ -79,19 +80,33 @@ func (opts *ConfigOpts) NewAlertConfiguration(projectID string) *admin.GroupAler
 	out.Enabled = &opts.enabled
 
 	if opts.matcherFieldName != "" {
+		if opts.matcherOperator == "" || opts.matcherValue == "" {
+			return nil, errors.New("--matcherOperator and --matcherValue are required when --matcherFieldName is provided")
+		}
 		out.Matchers = []map[string]interface{}{opts.newMatcher()}
 	}
 
 	if opts.metricThresholdMetricName != "" {
+		if opts.metricThresholdOperator == "" || opts.metricThresholdMode == "" || opts.metricThresholdUnits == "" || opts.metricThresholdThreshold == 0 {
+			return nil, errors.New("--metricOperator, --metricMode, --metricUnits and --metricThreshold are required when --metricName is provided")
+		}
 		out.MetricThreshold = opts.newMetricThreshold()
 	}
 
-	out.Notifications = []admin.AlertsNotificationRootForGroup{*opts.newNotification()}
+	if opts.notificationType == "" {
+		return nil, errors.New("--notificationType is required")
+	}
```

Comment:
why not mark the flag as required?

Reply:
We have `--file` support so we cannot make any other flag required.